client/capacitor/build.action.mjs

```diff
@@ -0,0 +1,131 @@
+import { fileURLToPath } from 'url';
+import path from 'path';
+import fs from 'fs/promises';
+import { spawn } from 'child_process';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname  = path.dirname(__filename);
+
+const capRoot    = __dirname;                // /.../client/capacitor (Capacitor config lives here)
+const clientRoot = path.resolve(capRoot, '..');     // /.../client
+const repoRoot   = path.resolve(clientRoot, '..');  // /.../outline-apps
+
+const isWin = process.platform === 'win32';
+const CAP_BIN_NAME = isWin ? 'cap.cmd' : 'cap';
+
+function bindir(dir) { return path.join(dir, 'node_modules', '.bin'); }
+async function exists(p) { try { await fs.access(p); return true; } catch { return false; } }
+
+function withLocalBins(extraEnv = {}) {
+  // Put both client/.bin and repo/.bin on PATH so either location works
+  const bins = [bindir(clientRoot), bindir(repoRoot)].join(path.delimiter);
+  const PATH = process.env.PATH ? `${bins}${path.delimiter}${process.env.PATH}` : bins;
+  return { ...process.env, PATH, ...extraEnv };
+}
```

Comment:
No need to parse the binary file path, `npx` will do the trick (as long as `@capacitor/cli` is declared as a dependency in `package-lock.json`). Please replace it with `spawnStream('npx', 'cap', ...)`